src/cfnlint/schemaManager.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+import platform
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+        self.boto3_sts = boto3.client('sts')
+        self.boto3_cfn = boto3.client('cloudformation')
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, name, registry_type):
+        account_id = self.boto3_sts.get_caller_identity().get('Account')
+        username = None
+        path_split = os.getcwd().split('/')
+        try:
+            index = path_split.index('Users')
+            username = path_split[index + 1]
+        except ValueError:
+            raise ValueError
+
+        is_windows = platform.system() == 'win32'
```

Comment:
📟  paging Windows expert @miparnisari / python expert @kddejong

thinking there might be some directory path logic that's more OS-agnostic and user-agnostic

is there some standard directory for Python packages that can be referenced instead?

Reply:
maybe something like these?

https://github.com/aws-cloudformation/cfn-lint/blob/29886f432989e4b32fa81f98e76208d740a857cb/src/cfnlint/helpers.py#L244

https://github.com/aws-cloudformation/cfn-lint/blob/29886f432989e4b32fa81f98e76208d740a857cb/src/cfnlint/maintenance.py#L53

https://github.com/aws-cloudformation/cfn-lint/blob/29886f432989e4b32fa81f98e76208d740a857cb/src/cfnlint/maintenance.py#L199

Reply:
Hmm, yea I wonder if this should be like this or not.  A lot of very OS and possible issues with specific configurations that has me a little concerned.